pkgs/applications/misc/coder/default.nix

```diff
@@ -0,0 +1,76 @@
+{ lib, pkgs, fetchFromGitHub, installShellFiles, buildGoModule, fetchYarnDeps }:
+
+let
+  pname = "coder";
+  version = "0.10.2";
+
+  src = fetchFromGitHub {
+    owner = "coder";
+    repo = "coder";
+    rev = "v${version}";
+    sha256 = "sha256-h5bN75agNocRAjShbufRCJr45huYJOzCBd4OcGpF4C4=";
+  };
+
+  offlineCache = fetchYarnDeps {
+    yarnLock = src + "/site/yarn.lock";
+    sha256 = "sha256-uDNPRQTpsgxyC5ks+2Qq/wiKjkbjWwSO+cJc5X6qmAA=";
+  };
+
+  yarn16 = pkgs.yarn.override { nodejs = pkgs.nodejs-16_x; };
+  nodePackages16 = pkgs.nodePackages.override { nodejs = pkgs.nodejs-16_x; };
+in
+buildGoModule rec {
+  inherit pname version src;
+
+  subPackages = [ "cmd/coder" ];
+
+  vendorSha256 = "sha256-+3Zy0zArCXkvD4ogfKdu9W9gJXveAhwFXKG1VRDvOkI=";
+
+  # Flags as provided by the build automation of the project:
+  #   https://github.com/coder/coder/blob/075e891f287b27cdb481a48e129f20a1e6a7de12/scripts/build_go.sh#L89
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/coder/coder/buildinfo.tag=${version}"
+  ];
+
+  preBuild = ''
```

Comment:
Can this be done in `postBuild`? If yes, it should be there; `pre*` hooks are usually used to prepare some files or fix something for the main portion of the phase.

Reply:
The project uses go embedd to embedd the build JS files